mirror_base.cpp

```diff
@@ -73,18 +73,21 @@ namespace mirror
 
 // ------------- !TOOLS -----------------
 
-	TypeSet g_typeSet;
 	TypeSet* GetTypeSet()
     {
+        static TypeSet g_typeSet;
         return &g_typeSet;
     }
 
 	TypeDesc* FindTypeByName(const char* _name)
 	{
-		return ::mirror::g_typeSet.findTypeByName(_name);
+		return GetTypeSet()->findTypeByName(_name);
 	}
 
 	MetaData::MetaData(const char* _name, const char* _data)
+    :
+    m_name(nullptr),
+    m_data(nullptr)
```

Comment:
What is the use of adding that ? the variables are initialized two lines after

Reply:
Oh yeah... I think I did that without thinking.